sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -117,14 +131,45 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
       case NonFatal(_) => None
     }
     val isTemp = sessionCatalog.isTempView(tableIdent)
+    val qualifier =
+      Array(metadata.map(_.identifier.database).getOrElse(tableIdent.database).orNull)
```

Comment:
`metadata.map(_.identifier.database).getOrElse(tableIdent.database).toArray`. In abnormal case we should return null, not Array(null)